backends/openvino/runtime/OpenvinoBackend.cpp

```diff
@@ -39,143 +39,159 @@ namespace backends {
-        throw std::runtime_error("OpenVINO runtime not available");
-    }
+  if (!is_available()) {
+    // ET_LOG(Error, "OpenVINO runtime is not available. Initialization
+    // failed.");
+    throw std::runtime_error("OpenVINO runtime not available");
+  }
 
-    //ET_LOG(Info, "OpenVINO runtime successfully verified and initialized.");
+  // ET_LOG(Info, "OpenVINO runtime successfully verified and initialized.");
 }
 
 bool OpenvinoBackend::is_available() const {
-    try {
-        // Create an OpenVINO Core object to verify runtime availability
-        ov::Core core;
-
-        // Check if at least one device is available
-        auto devices = core.get_available_devices();
-        if (!devices.empty()) {
-            return true;  // OpenVINO is available
-        }
-    } catch (const std::exception& e) {
-        // Log the exception if OpenVINO runtime is not available
-        ET_LOG(Error, "OpenVINO is not available: %s", e.what());
-    } catch (...) {
-        // Handle any unexpected errors
-        ET_LOG(Error, "OpenVINO availability check failed due to an unknown error.");
-    }
+  try {
+    // Create an OpenVINO Core object to verify runtime availability
+    ov::Core core;
 
-    return false;  // OpenVINO is not available
+    // Check if at least one device is available
+    auto devices = core.get_available_devices();
+    if (!devices.empty()) {
+      return true; // OpenVINO is available
+    }
+  } catch (const std::exception& e) {
+    // Log the exception if OpenVINO runtime is not available
+    ET_LOG(Error, "OpenVINO is not available: %s", e.what());
+  } catch (...) {
+    // Handle any unexpected errors
+    ET_LOG(
+        Error, "OpenVINO availability check failed due to an unknown error.");
+  }
+
+  return false; // OpenVINO is not available
 }
 
 Result<DelegateHandle*> OpenvinoBackend::init(
     BackendInitContext& context,
     FreeableBuffer* processed,
     ArrayRef<CompileSpec> compile_specs) const {
+  ET_LOG(Info, "OpenvinoBackend::init %p", processed->data());
 
-    ET_LOG(Info, "OpenvinoBackend::init %p", processed->data());
+  ov::Core core;
+  const char* data_ptr = static_cast<const char*>(processed->data());
+  size_t data_size = processed->size();
 
-    ov::Core core;
-    const char* data_ptr = static_cast<const char*>(processed->data());
-    size_t data_size = processed->size();
+  // Copy data to a string or vector
+  std::string data_string(data_ptr, data_size);
 
-    // Copy data to a string or vector
-    std::string data_string(data_ptr, data_size);
+  // Wrap the data in a stream
+  std::istringstream compiled_stream(data_string);
 
-    // Wrap the data in a stream
-    std::istringstream compiled_stream(data_string);
+  auto device = "CPU";
+  // Get the device value, if provided in compile sepcs
+  for (auto& compile_spec : compile_specs) {
+    if (std::strcmp(compile_spec.key, "device") == 0)
+      device = static_cast<char*>(compile_spec.value.buffer);
+  }
```

Comment:
Should we also check if the device is available or let it handled by "core.import_model" ?